### PR TITLE
Adds a preference for disabling "alternative" sounds

### PR DESCRIPTION
**Pull request by Absolucy**
:cl: Lucy
add: There is now a preference to disable "alternative" potentially misophonia-triggering sounds!
/:cl:

Currently affects the mentor noot, PDA glitch beep, and RPED "wooooosh"

### CODE REVIEW

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 1]
```diff
@@ -799,7 +799,7 @@ GLOBAL_LIST_EMPTY(PDAs)
 	tnote += signal
 
 	if (!silent)
-		if(HAS_TRAIT(SSstation, STATION_TRAIT_PDA_GLITCHED))
+		if(HAS_TRAIT(SSstation, STATION_TRAIT_PDA_GLITCHED) && can_play_alt_sound())
```
Comment (adamsong) [inline code/game/objects/items/devices/PDA/PDA.dm]: Why is the alt PDA sound based solely on the prefs of the person holding the PDA?

**Comment by Reed0506** [comment]
how could you do this to the beeps and the wooooosh

File: yogstation/code/datums/world_topic.dm [hunk 2]
```diff
@@ -80,7 +80,7 @@ GLOBAL_VAR_INIT(mentornoot, FALSE)
 	var/client/C = GLOB.directory[ckey(whom)]
 	if(!C)
 		return 0
-	if(GLOB.mentornoot || prob(1))
+	if((C.prefs.toggles & SOUND_ALT) && (GLOB.mentornoot || prob(1)))
```
Comment (adamsong) [inline yogstation/code/datums/world_topic.dm:83]:
```suggestion
	if(GLOB.mentornoot || (C.prefs.toggles & SOUND_ALT) && prob(1))
```

**Review by adamsong — APPROVED** [review]

**Review by adamsong — DISMISSED** [review]
That was supposed to be request

**Comment by Hopekz** [comment]
Just realized we have a sound file name of "pshoom_or_beepboopblorpzingshadashwoosh" since it's next to your changes

File: code/game/objects/items/devices/PDA/PDA.dm (hunk 1 above)
Comment (Absolucy) [reply code/game/objects/items/devices/PDA/PDA.dm]: hm, good idea, I could make an override in playsound

File: yogstation/code/datums/world_topic.dm (hunk 2 above)
Comment (Absolucy) [reply yogstation/code/datums/world_topic.dm:83]: no.

File: code/game/objects/items/devices/PDA/PDA.dm (hunk 1 above)
Comment (Absolucy) [reply code/game/objects/items/devices/PDA/PDA.dm]: i made it override using a list in playsound instead

**Comment by N3D6** [comment]
I really wish we'd stop with the "disabling this because I do not like it" prs

**Comment by maxion12345** [comment]
isn't miso a type of soup

**Comment by Absolucy** [comment]
@N3D6 it's pure QoL, meant to reduce annoyance, while not affecting gameplay in the slightest. i do not see the problem.

**Comment by adamsong** [comment]
It disables one of the was admins can torment mentors

**Comment by N3D6** [comment]
> It disables one of the was admins can torment mentors

This is why I am against this :kek:

**Comment by Marmio64** [comment]
let mentors give a stupid ass sound to ahelps so that we can see how they like being tortured

**Comment by adamsong** [comment]
If we gave mentors equality, they wouldn't be oppressed now would they?

**Comment by Marmio64** [comment]
> If we gave mentors equality, they wouldn't be oppressed now would they?

sounds good to me

**Comment by Hopekz** [comment]
So how does this work if the sound isn't sent directly to the player? Like for example the woosh sound since it is broadcasted.

For example for the woosh RPED: Does the player that has this preference off just never create that sound for others or does someone without the preference off still create the sound for you to hear despite you have it set to not hear them?

**Rewriting question to maybe better understand it if it was confusing:**
You: Have alt sounds disabled
Them: Have alt sounds enabled
Them: use the part device
The woosh sound gets created and broadcast on to the world

Do you still hear "woosh" since they created the noise?
Does this apply in reverse of you never being able to create the woosh sound yourself and no-one else hearing the woosh despite them wanting to have alt sounds enabled?

**Comment by Absolucy** [comment]
@Hopekz all sounds (unless sent via the `SEND_SOUND` macro) eventually boil down to a `playsound_local`, which is per-client, where we can then swap out sounds for individual clients.

**Comment by JamieD1** [comment]
Need to be careful here we dont encourage a mentality of catering to everyones likes and dislikes. People cant always get what they want. And I need you to confirm the Default is ON.

**Comment by Absolucy** [comment]
@JamieD1 

Default should be on: https://github.com/yogstation13/Yogstation/blob/40655d325382ea490f46c4d5f1482ab5cb01a46b/code/__DEFINES/preferences.dm#L30


Savefile upgrade should also turn it on automatically: https://github.com/yogstation13/Yogstation/blob/40655d325382ea490f46c4d5f1482ab5cb01a46b/code/modules/client/preferences_savefile.dm#L50-L51

File: yogstation/code/modules/mentor/mentor.dm [hunk 3]
```diff
@@ -104,7 +104,7 @@ GLOBAL_PROTECT(mentor_href_token)
 /proc/send_mentor_sound(client/C)
 	var/sound/pingsound = sound('yogstation/sound/misc/bikehorn_alert.ogg')
 	pingsound.volume = 90
-	if(GLOB.mentornoot || prob(10))
+	if((C.prefs.toggles & SOUND_ALT) && (GLOB.mentornoot || prob(1)))
```
Comment (Bibby0110) [inline yogstation/code/modules/mentor/mentor.dm]:
```suggestion
	if((C.prefs.toggles & SOUND_ALT) && (GLOB.mentornoot || prob(10)))
```

Comment (Absolucy) [reply yogstation/code/modules/mentor/mentor.dm]: why? it was 1 before

Comment (adamsong) [reply yogstation/code/modules/mentor/mentor.dm]: Except it wasn't, check the changes on the PR

Comment (Absolucy) [reply yogstation/code/modules/mentor/mentor.dm]: oh whoops, the other one was `prob(1)`, you're right

**Review by JamieD1 — APPROVED** [review]

**Pull request merged**